### PR TITLE
sql: Add transaction statement

### PR DESCRIPTION
**Pull request by shenli**
All things about transaction in TiDB.

### CODE REVIEW

File: sql/transaction.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+```
+提交当前事务，包括从 `BEGIN` 到 `COMMIT` 之间的所有修改。
+
+
+## ROLLBACK
+语法：
+```sql
+ROLLBACK;
+```
+回滚当前事务，撤销从 `BEGIN` 到 `ROLLBACK` 之间地所有修改。
```
Comment (queenypingcap) [inline sql/transaction.md]: 之间地所有 -> 之间的所有

**Comment by shenli** [comment]
@queenypingcap PTAL

**Comment by shenli** [comment]
@sunhao2017

File: sql/transaction.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+title: 事务语言
```
Comment (andelf) [inline sql/transaction.md]: 语言 or 语句

File: sql/transaction.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## ROLLBACK
+语法：
+```sql
+ROLLBACK;
+```
+回滚当前事务，撤销从 `BEGIN` 到 `ROLLBACK` 之间的所有修改。
+
+
+## 显式事务和隐式事务
+TiDB 可以显式或隐式地使用事务（`BEGIN/COMMIT`） 或者隐式的使用事务 （`SET autocommit = 1`）。
```
Comment (andelf) [inline sql/transaction.md]: wrong semantic

**Comment by shenli** [comment]
@andelf PTAL

**Pull request merged**